astroplan/plots/core.py

```diff
@@ -0,0 +1,223 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+import matplotlib.pyplot as plt
+import numpy as np
+import astropy.units as u
+import random
+
+
+def plot_airmass(target, observer, time, ax=None, style_kwargs=None,
+                 dark_plot=False):
+    """
+    TODO:
+        1) Timezones?
+        2) Limit airmass <= 3.
+        3) Dark plot option.
+        4) Dealing with no target name.
+        5) ax.figure.autofmt_xdate() ?
+
+    Plots airmass as a function of time for a given target.
+
+    If an ax object already exists, an additional airmass plot will be
+    "stacked" on it.
+    Otherwise, creates a new ax object and plots airmass on top of that.
+
+    When a scalar `Time` object is passed in (e.g., Time('2000-1-1')), the
+    resulting plot will use a 24-hour window centered on the time indicated,
+    with airmass sampled at regular intervals throughout.
+    However, the user can control the exact number and frequency of airmass
+    calculations used by passing in a non-scalar `Time` object. For instance,
+    Time(['2000-1-1 23:00:00', '2000-1-1 23:30:00']) will result in a plot with
+    only two airmass measurements.
+
+    Parameters
+    ----------
+    target : `~astroplan.FixedTarget`
+        The celestial body of interest.
+
+    observer : `~astroplan.Observer`
+        The person, telescope, observatory, etc. doing the observing.
+
+    time : `~astropy.time.Time`
+        Can be scalar (e.g., Time('2000-1-1')) or not (e.g., Time(['2000-1-1'])).
+
+    ax : `~matplotlib.axes.Axes` or None, optional.
+        The axes object to be drawn on.
+        If None, use the current axes (matplotlib.pyplot.gca).
+
+    style_kwargs : dict or Empty, optional.
+        A dictionary of keywords passed into `matplotlib.pyplot.plot_date`
+        to set plotting styles.
+
+    dark_plot : bool, optional.
```

Comment:
If we're not going to tackle this in this PR (which I am totally fine with!), please remove the `dark_plot` kwarg since we're not totally sure this will even exist when it is implemented.